### PR TITLE
Create shared by link page

### PR DESCRIPTION
**Pull request by LukasHirt**
We've added a new page called "Shared via link". This page displays a files list containing only resources shared via public links.

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/13977/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/13977/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14009/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14009/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14009/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by wkloucek** [comment]
@LukasHirt please switch to `owncloudci/golang:1.16`

**Comment by LukasHirt** [comment]
> please switch to owncloudci/golang:1.16

I'll do that, thanks! @wkloucek

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14014/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14016/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14018/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14018/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14020/

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14020/

<p>









</p></details>

**Comment by LukasHirt** [comment]
Reverted back to `webhippie` as the `owncloudci` golang images are breaking the CI currently. Need to be taken care of in own PR

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14034/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14038/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14038/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14041/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14041/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by LukasHirt** [comment]
Tests failing in the CI but not locally (will be updating this comment in case of new ones):

- [  ] webUISharingExternalToRoot/federationSharing.feature:250

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14043/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRootSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14047/

<p>




</p></details>

File: packages/web-app-files/src/views/SharedViaLink.vue [hunk 1]
```diff
@@ -0,0 +1,182 @@
+<template>
+  <div>
+    <list-loader v-if="loading" />
+    <template v-else>
+      <no-content-message
+        v-if="isEmpty"
+        id="files-shared-with-others-empty"
```
Comment (kulmann) [inline packages/web-app-files/src/views/SharedViaLink.vue:7]:
```suggestion
        id="files-shared-via-link-empty"
```

File: packages/web-app-files/src/views/SharedViaLink.vue [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        class="files-empty"
+        icon="group"
+      >
+        <template #message>
+          <span v-translate>There are no resources with a public link at the moment</span>
+        </template>
+      </no-content-message>
+      <oc-table-files
+        v-else
+        id="files-shared-with-others-table"
```
Comment (kulmann) [inline packages/web-app-files/src/views/SharedViaLink.vue:17]:
```suggestion
        id="files-shared-via-link-table"
```

**Review by kulmann — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14047/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRootSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14052/

<p>




</p></details>

**Comment by LukasHirt** [comment]
@individual-it Could anybody from your team pls update the screenshot in oCIS for `webUIFiles/fileList.feature:40`? We currently have issues running acceptance tests for oCIS on mac so I cannot updated them there. I tried to copy the screenshot from oC10 but oCIS hasn't got Favorites item 🙈

**Review by kulmann — APPROVED** [review]
Looks good to me 👍  baseline image of the sidebar for ocis needs to be solved.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14049/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14052/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by individual-it** [comment]
we are tracking random failures currently here https://github.com/owncloud/web/issues/4878
I've added  webUISharingInternalGroupsToRootSharingIndicator/shareWithGroups.feature:92

 webUISharingPublicDifferentRoles/shareByPublicLinkDifferentRoles.feature:92 was already on the list

and I could not find any drone run where webUISharingExternalToRoot/federationSharing.feature:250 failed

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14071/

<p>





</p></details>

**Comment by individual-it** [comment]
deletion has still a problem with locked files on oC10. Tests will be skipped in https://github.com/owncloud/web/pull/4897

see also
  https://github.com/owncloud/web/issues/4582
  https://github.com/owncloud/web/issues/4784

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRootSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14089/

<p>




</p></details>

**Pull request merged**